app/ts/Elicitation/RankingElicitation/OrdinalRankingUtil.ts

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash';
+import {UNRANKED} from '../elicitationConstants';
+import IElicitationCriterion from '../Interface/IElicitationCriterion';
+import IOrdinalRanking from '../Interface/IOrdinalRanking';
+import IRankingAnswer from '../Interface/IRankingAnswer';
+
+export function findCriterionIdForRank(
+  criteria: Record<string, IElicitationCriterion>,
+  rankings: Record<string, IRankingAnswer>,
+  rank: number
+): string {
+  return _.find(criteria, (criterion) => {
+    return rankings[criterion.id] && rankings[criterion.id].rank === rank;
+  }).id;
+}
+
+export function assignMissingRankings(
+  rankings: Record<string, IRankingAnswer>,
+  selectedCriterionId: string,
+  rank: number,
+  criteria: Record<string, IElicitationCriterion>
+): Record<string, IRankingAnswer> {
+  let finishedRankings = _.cloneDeep(rankings);
```

Comment:
it is tricky because the update and assigning the last rank would take place in the body of the same function -> the rankings will not be yet updated when they reach utils

Reply:
I would propose to move `setRanking` from context to util, so that at least that duplication is gone